File: api/app/domain/models/tool_result.py
from typing import Generic, Optional, TypeVar
from pydantic import BaseModel


T = TypeVar("T")


class ToolResult(BaseModel, Generic[T]):
    """工具结果Domain模型"""

    success: bool = True
    data: Optional[T] = None  # 工具执行结果/数据
    message: Optional[str] = None  # 额外信息提示

    @classmethod
    def from_sandbox(
        cls, code: int, message: str, data: Optional[T], **kwargs
    ) -> "ToolResult":
        return cls(success=True if code < 300 else False, message=message, data=data)

File: api-node/src/domain/tool-result.ts
export interface ToolResult<T = unknown> {
  success: boolean;
  message?: string;
  data?: T;
  error?: string;
}

export function toolSuccess<T>(data?: T, message?: string): ToolResult<T> {
  return { success: true, data, message };
}

export function toolFailure(message: string, error?: string): ToolResult {
  return { success: false, message, error: error ?? message };
}
